include/zephyr/dsp/dsp.h

```diff
@@ -17,6 +17,14 @@
 #define DSP_FUNC_SCOPE
 #endif
 
+#ifdef CONFIG_DSP_BACKEND_ARC
+#define DSP_DATA __agu
```

Comment:
I think that as we get more and more architecture backends we'll get more of these attributes. Can we use the same pattern as the static and have this as a Kconfig without a prompt in case ? maybe `config DSP_BACKEND_HAS_AGU`? Similarly for the data attribute, we can have a string Kconfig for the section. My goal is to make as few custom (per architecture) switches.

Reply:
I'm still a bit lost about the scalability of this approach, can you explain in more details how you see this when the next attribute comes up?

So lets assume we have `CONFIG_DSP_BACKEND_HAS_AGU` and a new attribute `CONFIG_DSP_BACKEND_HAS_X`, does the code now look like:

```
#if defined(CONFIG_DSP_BACKEND_HAS_AGU) && defined(CONFIG_DSP_BACKEND_HAS_X)
#define DSP_DATA __agu __x
#elif defined(CONFIG_DSP_BACKEND_HAS_AGU) && !defined(CONFIG_DSP_BACKEND_HAS_X)
#define DSP_DATA __agu
#elif !defined(CONFIG_DSP_BACKEND_HAS_AGU) && defined(CONFIG_DSP_BACKEND_HAS_X)
#define DSP_DATA __x
#endif
```

Reply:
I'd expect these would be mutual exclusive with at most a single attribute at a time. The _HAS refers to a specific hardware feature or flavor of DSP acceleration, so I don't expect a combination of _HAS_AGU and _HAS_X likely.
But now I understand your scalability concern better.  Can we address it in the future when the case of HAS_AGU && HAS_X occurs?

Reply:
I also expect these are mutual exclusive, so the code shouldn't be such complex. I tried to use string Kconfig to directly pass attributes to C macro `DSP_DATA` and `DSP_STATIC_DATA` so that there is no if statement in code. Kconfig and C macro definition might look like:
```
config DSP_DATA
	string
	default "__agu" if DSP_BACKEND_ARCMWDT
	default "__x" if DSP_BACKEND_X
```
`#define DSP_DATA CONFIG_DSP_DATA`
But I found double quotes are introduced to C macro by make itself, which is unrecognizable by compiler. If this can be solved, the code can be simple.